Calling find_package(ament_cmake REQUIRED) multiple times during a build fails

I have a CMake build that builds several ament packages. Currently each subproject calls

`find_package(ament_cmake REQUIRED)`

Which fails the build with this error message:

```
CMake Error at ~/ros2-linux/share/ament_cmake_core/cmake/ament_cmake_uninstall_target-extras.cmake:35 (add_custom_target):
  add_custom_target cannot create target "uninstall" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "~/Jenkins/workspace/Branch-OI7ZPASJZEV3XGQWBFQY2PEQXIELKYENI5G6JZQZXQR3CPC6F5NA@2/tests/eclipseProject/algos/testsProto/subscriber/ros".
  See documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  ~/ros2-linux/share/ament_cmake_core/cmake/ament_cmake_coreConfig.cmake:30 (include)
  ~/ros2-linux/share/ament_cmake/cmake/ament_cmake_export_dependencies-extras.cmake:15 (find_package)
  ~/ros2-linux/share/ament_cmake/cmake/ament_cmakeConfig.cmake:30 (include)
  algos/testsProto/subscriberDefault/ros/CMakeLists.txt:16 (find_package)
```

I would expect the amen_cmakeConfig.cmake to check whether ament_cmake was already found, and in that case not do anything, or something like that.

Version is 0.4.0

--- a/ament_cmake_core/ament_cmake_uninstall_target-extras.cmake
+++ b/ament_cmake_core/ament_cmake_uninstall_target-extras.cmake
@@ -31,7 +31,7 @@ if(AMENT_CMAKE_UNINSTALL_TARGET)
     @ONLY
   )
 
-# register uninstall target to run generated CMake script
-add_custom_target(${PROJECT_NAME}_uninstall
-  COMMAND ${CMAKE_COMMAND} -P "${AMENT_CMAKE_UNINSTALL_TARGET_UNINSTALL_SCRIPT}")
+  # register uninstall target to run generated CMake script
+  add_custom_target(${PROJECT_NAME}_uninstall
+    COMMAND ${CMAKE_COMMAND} -P "${AMENT_CMAKE_UNINSTALL_TARGET_UNINSTALL_SCRIPT}")
 endif()

--- a/ament_cmake_core/ament_cmake_uninstall_target-extras.cmake
+++ b/ament_cmake_core/ament_cmake_uninstall_target-extras.cmake
@@ -31,7 +31,12 @@ if(AMENT_CMAKE_UNINSTALL_TARGET)
     @ONLY
   )
 
+  if (NOT TARGET uninstall)
+    add_custom_target(uninstall)
+  endif()
+
   # register uninstall target to run generated CMake script
   add_custom_target(${PROJECT_NAME}_uninstall
     COMMAND ${CMAKE_COMMAND} -P "${AMENT_CMAKE_UNINSTALL_TARGET_UNINSTALL_SCRIPT}")
+  add_dependencies(uninstall ${PROJECT_NAME}_uninstall)
 endif()

--- a/ament_cmake_core/ament_cmake_uninstall_target-extras.cmake
+++ b/ament_cmake_core/ament_cmake_uninstall_target-extras.cmake
@@ -31,7 +31,7 @@ if(AMENT_CMAKE_UNINSTALL_TARGET)
     @ONLY
   )
 
-  # register uninstall target to run generated CMake script
-  add_custom_target(uninstall
-    COMMAND ${CMAKE_COMMAND} -P "${AMENT_CMAKE_UNINSTALL_TARGET_UNINSTALL_SCRIPT}")
+# register uninstall target to run generated CMake script
+add_custom_target(${PROJECT_NAME}_uninstall
+  COMMAND ${CMAKE_COMMAND} -P "${AMENT_CMAKE_UNINSTALL_TARGET_UNINSTALL_SCRIPT}")
 endif()

